homeassistant/components/melnor/sensor.py

```diff
@@ -98,5 +161,31 @@ def __init__(
 
     @property
     def native_value(self) -> StateType:
-        """Return the battery level."""
+        """Return the sensor value."""
         return self.entity_description.state_fn(self._device)
+
+
+class MelnorZoneSensorEntity(MelnorZoneEntity, SensorEntity):
+    """Representation of a Melnor sensor."""
+
+    entity_description: MelnorZoneSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: MelnorDataUpdateCoordinator,
+        entity_description: MelnorZoneSensorEntityDescription,
+        valve: Valve,
+    ) -> None:
+        """Initialize a sensor for a Melnor device."""
+        super().__init__(coordinator, valve)
+
+        self._attr_unique_id = (
+            f"{self._device.mac}-zone{valve.id}-{entity_description.key}"
+        )
+
+        self.entity_description = entity_description
```

Comment:
It looks like this code is shared with `number.py` 

Can you move it into a base class?